tests/test_server.py

```diff
@@ -52,140 +71,152 @@ def test_add_device(client, name='steelseries-rival310'):
     data = {
         'shortname': name
     }
-    response = client.post('/devices/add',
-                           data=json.dumps(data),
-                           content_type='application/json')
+    response = client.post('/devices/add', json=data)
```

Comment:
before that you should probably ensure that the server has no device named `name`, and then check after the `assert` below that the device is part of the `list_devices` API.

Reply:
Why? We can have multiple devices with the same name. That's why we an ID.

Reply:
it would make sense to have unique names, especially during the tests.
But the whole point of testing that `add a new device` works consists in:
- make sure you do not have the device before hand
- add the device
- then check that the device has been properly added

for the first point, I suggested to make sure you do not have a `name` already, because tests might be running in parallel. Ideally, you should ensure that `list_device` returns an empty list, or even better that no event node exists with that particular unique name.

The ID is nice, but for what we know, there is no guarantees that the server will not just fake a new device, and assign a new id to an existing one.

Reply:
> The ID is nice, but for what we know, there is no guarantees that the server will not just fake a new device, and assign a new id to an existing one.

The current implementation does not do this, and I can document this behavior in the API documentation. Would that be enough? I really don't like that approach, I think IDs make more sense, but if you share the same opinion I guess I could change it.

Reply:
> The current implementation does not do this

That's exactly the point of regression tests: you do not trust the underlying code, and you write it as if it was faulty

>  I can document this behavior in the API documentation. Would that be enough?

No, unless you have a test that ensure that the ID is unique, which sums up to exactly what I suggested previously